c7n/reports/csvout.py

```diff
@@ -62,24 +62,33 @@
 log = logging.getLogger('custodian.reports')
 
 
-def report(policy, start_date, options, output_fh, raw_output_fh=None):
+def report(policies, start_date, options, output_fh, raw_output_fh=None):
     """Format a policy's extant records into a report."""
     formatter = Formatter(
-        policy.resource_manager,
+        policies[0].resource_manager,
         extra_fields=options.field,
         no_default_fields=options.no_default_fields,
+        multipolicy=len(policies) > 1
     )
 
-    if policy.ctx.output.use_s3():
-        records = record_set(
-            policy.session_factory,
-            policy.ctx.output.bucket,
-            policy.ctx.output.key_prefix,
-            start_date)
-    else:
-        records = fs_record_set(policy.ctx.output_path, policy.name)
+    records = []
+    for policy in policies:
+        if policy.ctx.output.use_s3():
+            policy_records = record_set(
+                policy.session_factory,
+                policy.ctx.output.bucket,
+                policy.ctx.output.key_prefix,
+                start_date)
+        else:
+            policy_records = fs_record_set(policy.ctx.output_path, policy.name)
+
+        log.debug("Found %d records for region %s", len(policy_records), policy.options.region)
 
-    log.debug("Found %d records", len(records))
+        for record in policy_records:
+            record['policy'] = policy.name
```

Comment:
@kapilt - fixed for both cases.  It will only include the `region` column if there are multiple regions in the output, and it will only include the `policy` column if there are multiple unique policy names